### ISSUE TITLE
Implement autoenv unload

### ISSUE THREAD

**Issue by turicas** [hyperupcall/autoenv#30]
In general I use autoenv to automatically activate my Python virtualenvs. It works pretty nice, but I miss a feature to automatically deactivate the virtualenv. It should deactivate the virtualenv if I change the directory to a path that is not inside the path which enabled the virtualenv. For example:

``` bash
mkdir spam
echo "workon spam" > spam/.env
echo "deactivate" > spam/.env-unload
cd spam # run `source .env`, activating virtualenv 'spam'

# working in this directory...
mkdir eggs
cd eggs
touch ham
cd ..
echo "finished" > README.markdown
# finished working.

cd .. # runs `source .env-unload`, deactivating virtualenv
```

Another approach is to define bash functions inside `.env` (like `autoenv_load` and `autoenv_unload`) and call these functions instead of just sourcing `.env`/`.env-unload`.

**Comment by msjaiswal** [comment]
+1

**Comment by blueyed** [comment]
FYI: [zsh-autoenv](https://github.com/Tarrasch/zsh-autoenv) also supports this, either through a separate or the same file (where you can look at a variable to see which event it is).

**Comment by unforswearing** [comment]
+1

**Comment by kevinpostal** [comment]
+1

**Comment by Atry** [comment]
+1

**Comment by luizs81** [comment]
+1

**Comment by aliavni** [comment]
+1

**Comment by SashaOv** [comment]
I would suggest documenting the workaround by @kazagistar above somewhere in README. It could be useful for many people

**Comment by yuanbohan** [comment]
+1

**Comment by wddabc** [comment]
Hi, I'm running into the same issue and found the suggestion by @kazagistar to be very useful. Thanks.

One minor issue of that snippet is it will always run whenever I enter the subfolders, which will unnecessarily deactivate and reactivate the env. 
My two cents on the improvement:

```
recent_env=$(pwd)
while [ "${recent_env}" != "$HOME" ];do
    if [ -f ${recent_env}/.env ];then
        return
    fi
    recent_env=$(dirname "${recent_env}")
done
source deactivate
``` 
This will make sure the `source deactivate` runs only when there are no other .env files along the path except the one on the $HOME.

**Comment by kennethreitz** [comment]
I recommend just using direnv for this functionality.

**Comment by hyperupcall** [comment]
Closed as this was implemented in [v0.3.0](https://github.com/hyperupcall/autoenv/releases/tag/v0.3.0). Sorry for the pings - I know this is an older thread. Feel free make a new issue for bug reports / unexpected functionality.
